logs/field.go

```diff
@@ -0,0 +1,260 @@
+package logs
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// Field represents typed log field (a key-value pair). Adapters should determine
+// Field's type based on Type and use the corresponding getter method to retrieve
+// the value:
+//
+//	switch f.Type() {
+//	case logs.IntType:
+//	    var i int = f.Int()
+//	    // handle int value
+//	case logs.StringType:
+//	    var s string = f.String()
+//	    // handle string value
+//	//...
+//	}
+//
+// Getter methods must not be called on fields with wrong Type (e.g. calling String()
+// on fields with Type != StringType).
+// Field must not be initialized directly as a struct literal.
+type Field struct {
+	ftype FieldType
+	key   string
+
+	vint int64
+	vstr string
+	vany interface{}
+}
+
+func (f Field) Type() FieldType {
+	return f.ftype
+}
+
+func (f Field) Key() string {
+	return f.key
+}
+
+// String is a value getter for fields with StringType type
+func (f Field) String() string {
+	return f.vstr
+}
+
+// Int is a value getter for fields with IntType type
+func (f Field) Int() int {
+	return int(f.vint)
```

Comment:
What about check f.Type and panic if call Int() for bad time (for example String).

That allow to see bad values used and will consistent with other types combination, for example Strings and Stringer - it will panic while cast interface.